01_fib/CameronSchafer/fib.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+//base fibonacci function
+func fib(n int) {
+	sign, n1, n2 := 1, 0, 1
+	//check if use nega or normal fib sequence
+	if n < 0 {
+		sign = -1
+	}
+	//loop until all numbers have been printed
+	for i := 0; i < n*sign; i++ {
```

Comment:
We could potentially optimise this by removing the multiplication everytime the the condition is evaluated.  Calculating the sequence length once in advance given the value `n*sign` does not change during iteration.

Reply:
Thank you, I will try change this up now.

Reply:
@nicholascross I updated the code as per your recommendation.